### PR TITLE
improve code coverage

### PR DESCRIPTION
**Pull request by jackiemunroe**
Increased code coverage and testability for:
* hub/vendor/customer.py
* sub/customer.py

https://jira.mozilla.com/browse/COPS-827

Note: the coverage report generated by `doit test` shows 99% for vendor/customer.py. However, I have verified that the lines that are being shown as missed are indeed being executed by running individual tests manually.

### CODE REVIEW

File: src/sub/tests/unit/test_customer.py [hunk 1]
```diff
@@ -1,49 +1,255 @@
...
+from sub.shared.db import SubHubAccountModel
+from sub.shared.cfg import CFG
+from sub.shared.exceptions import IntermittentError, ServerError
+
+
+class CustomerTest(TestCase):
+    def setUp(self) -> None:
+
+        with open("tests/unit/fixtures/stripe_cust_test1.json") as fh:
+            cust_test1 = json.loads(fh.read())
```
Comment (marty331) [inline src/sub/tests/unit/test_customer.py]: Should close file after read - fh.close()

File: src/sub/tests/unit/test_customer.py [hunk 2]
```diff
@@ -1,49 +1,255 @@
...
+        self.customer1 = convert_to_stripe_object(cust_test1)
+
+        with open("tests/unit/fixtures/stripe_cust_test2.json") as fh:
+            cust_test2 = json.loads(fh.read())
```
Comment (marty331) [inline src/sub/tests/unit/test_customer.py]: Should close file after read - fh.close()

File: src/sub/tests/unit/test_customer.py [hunk 3]
```diff
@@ -1,49 +1,255 @@
...
+        self.customer2 = convert_to_stripe_object(cust_test2)
+
+        with open("tests/unit/fixtures/stripe_cust_test4_no_subs.json") as fh:
+            cust_test1_no_subs = json.loads(fh.read())
```
Comment (marty331) [inline src/sub/tests/unit/test_customer.py]: Should close file after read - fh.close()

File: src/sub/tests/unit/test_customer.py [hunk 4]
```diff
@@ -1,49 +1,255 @@
...
+        self.customer_no_subs = convert_to_stripe_object(cust_test1_no_subs)
+
+        with open("tests/unit/fixtures/stripe_cust_test5_no_sources.json") as fh:
+            cust_test1_no_sources = json.loads(fh.read())
```
Comment (marty331) [inline src/sub/tests/unit/test_customer.py]: Should close file after read - fh.close()

File: src/sub/tests/unit/test_customer.py [hunk 5]
```diff
@@ -1,49 +1,255 @@
...
+        self.customer_no_sources = convert_to_stripe_object(cust_test1_no_sources)
+
+        with open("tests/unit/fixtures/stripe_deleted_cust.json") as fh:
+            deleted_cust = json.loads(fh.read())
```
Comment (marty331) [inline src/sub/tests/unit/test_customer.py]: Should close file after read - fh.close()

File: src/sub/tests/unit/test_customer.py [hunk 6]
```diff
@@ -1,49 +1,255 @@
...
+        self.deleted_customer = convert_to_stripe_object(deleted_cust)
+
+        with open("tests/unit/fixtures/stripe_cust_list.json") as fh:
+            cust_list = json.loads(fh.read())
```
Comment (marty331) [inline src/sub/tests/unit/test_customer.py]: Should close file after read - fh.close()

File: src/sub/tests/unit/test_customer.py [hunk 7]
```diff
@@ -1,49 +1,255 @@
...
+
```
Comment (marty331) [inline src/sub/tests/unit/test_customer.py]: Why the blank line?

File: src/sub/tests/unit/test_customer.py [hunk 8]
```diff
@@ -1,49 +1,255 @@
...
+        self.empty_customer_list = convert_to_stripe_object(cust_list)
+
+        cust_list["data"].append(cust_test2)
+        cust_list["data"].append(cust_test1)
+        self.customer_list = convert_to_stripe_object(cust_list)
+
+        with open("tests/unit/fixtures/stripe_sub_test1.json") as fh:
+            sub_test1 = json.loads(fh.read())
```
Comment (marty331) [inline src/sub/tests/unit/test_customer.py]: Should close file after read - fh.close()

File: src/hub/tests/unit/stripe/customer/test_stripe_customer.py [hunk 9]
```diff
@@ -2,199 +2,538 @@
...
-    filename = "customer-subscription-deleted.json"
-    run_customer(mocker, data, filename)
-    unstub()
+from hub.shared.db import SubHubDeletedAccountModel
+
+
+class StripeCustomerCreatedTest(TestCase):
+    def setUp(self) -> None:
+        with open("tests/unit/fixtures/stripe_cust_created_event.json") as fh:
+            self.customer_created_event = json.loads(fh.read())
```
[the remark this answers is not in the document]
Comment (jackiemunroe) [reply src/hub/tests/unit/stripe/customer/test_stripe_customer.py]: By using "with" the file that is opened is automatically closed. This was introduced in Python 2.4 https://docs.python.org/2.5/whatsnew/pep-343.html

[the remark this answers is not in the document]
Comment (jackiemunroe) [reply src/hub/tests/unit/stripe/customer/test_stripe_customer.py]: > It is good practice to use the with keyword when dealing with file objects. The advantage is that the file is properly closed after its suite finishes, even if an exception is raised at some point.

https://docs.python.org/3.7/tutorial/inputoutput.html#reading-and-writing-files

[the remark this answers is not in the document]
Comment (marty331) [reply src/hub/tests/unit/stripe/customer/test_stripe_customer.py]: I've always closed the file as a best practice but I will let this go.

**Pull request merged**